src/integrationTest/java/com/pubnub/api/integration/HistoryIntegrationTest.java

```diff
@@ -146,22 +158,57 @@ public void testFetchSingleChannel_Meta() throws PubNubException {
 
         publishMixed(pubNub, 10, expectedChannelName);
 
-        pause(3);
-
-        final PNFetchMessagesResult fetchMessagesResult = pubNub.fetchMessages()
-                .channels(Collections.singletonList(expectedChannelName))
-                .maximumPerChannel(25)
-                .includeMeta(true)
-                .sync();
+        final PNFetchMessagesResult fetchMessagesResult = pauseUntilAsserted(() -> {
+            PNFetchMessagesResult result = pubNub.fetchMessages()
+                    .channels(Collections.singletonList(expectedChannelName))
+                    .maximumPerChannel(25)
+                    .includeMeta(true)
+                    .sync();
+            assertNotNull(result);
+            assertThat(result.getChannels().size(), greaterThan(0));
+            return result;
+        });
 
-        assert fetchMessagesResult != null;
         for (PNFetchMessageItem messageItem : fetchMessagesResult.getChannels().get(expectedChannelName)) {
             assertNotNull(messageItem.getMessage());
             assertNotNull(messageItem.getTimetoken());
             assertNotNull(messageItem.getMeta());
             assertNull(messageItem.getActions());
         }
+    }
+
+    @Test
+    public void testFetchSingleChannel_SpaceId_MessageType() throws PubNubException {
+        final String expectedChannelName = random();
+        final String message = "message";
+        final String expectedSpaceIdValue = "mySpace";
```

Comment:
You could do the same trick with SpaceId as with the MessageType and use the type across the test